### DISCUSSION TITLE
Consider using ariadne

### DISCUSSION THREAD

**Discussion by tristan-f-r** [pest-parser/pest#772] [Ideas]
https://github.com/zesterer/ariadne is a great error diagnostics library that produces beautiful errors up to the standards of even rust's error reporting system. It handles a lot of edge cases and terminal backwards compatibility.

**Comment by tomtau** [comment]
I think this could potentially be added under a feature flag. FYI there was a previous WIP work to add [miette](https://github.com/zkat/miette): https://github.com/pest-parser/pest/pull/663 which can produce similarly formatted error messages.

**Comment by tomtau** [comment]
this was just added: https://github.com/pest-parser/pest/pull/1038
